### ISSUE TITLE
Extend `pytest-raises-with-multiple-statements (PT012)` to cover `pytest.warns`

### ISSUE THREAD

**Issue by harupy** [astral-sh/ruff#14239]
[`pytest-raises-with-multiple-statements`](https://docs.astral.sh/ruff/rules/pytest-raises-with-multiple-statements/) only covers `pytest.raises`. It'd be nice if it covers `pytest.warns` as well.

```python
import pytest


def test_raises():
    with pytest.raises(Exception):  # PT012
        f3()
        f4()


def test_warns():
    with pytest.warns(UserWarning):
        f1()  # does this warn?
        f2()  # or perhaps this one warns?
        # -> Not obvious which line warns.
```

Playground to ensure `pytest.warns` is not covered: https://play.ruff.rs/271e2f8e-751a-4368-971c-322eeeb10b39

References:

- https://docs.pytest.org/en/stable/reference/reference.html#pytest.warns

**Comment by harupy** [comment]
A new rule might make more sense to avoid confusion.

**Comment by harupy** [comment]
Adding a new rule is another option.

**Comment by harupy** [comment]
> I think that would have to be its own rule unless we rename the existing rule

Makes sense.

**Comment by harupy** [comment]
@MichaReiser do we need to add the same rule in https://github.com/m-burst/flake8-pytest-style to add it in ruff?

**Comment by MichaReiser** [comment]
I think it should be fine. We just need to make sure that we use a rule code that's unlikely to collide with any new rule added to the upstream plugin

**Comment by tjkuson** [comment]
Created an issue in the upstream repo: https://github.com/m-burst/flake8-pytest-style/issues/317

**Comment by MichaReiser** [comment]
Thanks @tjkuson

**Comment by snowdrop4** [comment]
This lint makes a lot of sense to me.

I'll make a draft PR.

I'll just use a placeholder code until flake8 decides on one.

**Comment by snowdrop4** [comment]
I have a finished branch: https://github.com/snowdrop4/ruff/tree/AVK/PytestWarnsWithMultipleStatements

But yeah, I'll wait and see what the flake8 plugin people say before PRing.

**Comment by MichaReiser** [comment]
The upstream issue has been closed as completed in case someone's interested to work on this

**Comment by tjkuson** [comment]
I volunteer unless @snowdrop4 fancies it
